### PR TITLE
Make param 3 optional and skip if it's not available for WP compat

### LINKED ISSUE #18628: [3.3 RC 2] Error: Too few arguments to function WC_Regenerate_Images_Request::filter_image_sizes_to_only_missing_thumbnails()
Logging (after changing thumbnail cropping settings):

```
PHP Fatal error:  Uncaught ArgumentCountError: Too few arguments to function WC_Regenerate_Images_Request::filter_image_sizes_to_only_missing_thumbnails(), 2 passed in ~\wp-includes\class-wp-hook.php on line 286 and exactly 3 expected in ~\wp-content\plugins\woocommerce\includes\class-wc-regenerate-images-request.php:149
Stack trace:
#0 ~\wp-includes\class-wp-hook.php(286): WC_Regenerate_Images_Request->filter_image_sizes_to_only_missing_thumbnails(Array, Array)
#1 ~\wp-includes\plugin.php(203): WP_Hook->apply_filters(Array, Array)
#2 ~\wp-admin\includes\image.php(129): apply_filters('intermediate_im...', Array, Array)
#3 ~\wp-content\plugins\woocommerce\includes\class-wc-regenerate-images-request.php(115): wp_generate_attachment_metadata(2088, '~')
#4 ~\wp-content\plugins\woocommerce\includes\libraries\wp-background-proce in ~\wp-content\plugins\woocommerce\includes\class-wc-regenerate-images-request.php on line 149
```

### PR DESCRIPTION
**Pull request by bor0**
Fixes #18628.

### CODE REVIEW

**Comment by WPprodigy** [comment]
Should prolly exit earlier if argument 3 isn't there, since line 154 uses it.

File: includes/class-wc-regenerate-images-request.php [hunk 1]
```diff
@@ -146,15 +146,15 @@ protected function task( $item ) {
...
+	public function filter_image_sizes_to_only_missing_thumbnails( $sizes, $metadata, $attachment_id = null ) {
 		if ( ! $sizes ) {
 			return $sizes;
 		}
 
 		$fullsizepath = get_attached_file( $attachment_id );
 		$editor       = wp_get_image_editor( $fullsizepath );
 
-		if ( is_wp_error( $editor ) ) {
+		if ( is_wp_error( $editor ) || is_null( $attachment_id ) ) {
```
Comment (kloon) [inline includes/class-wc-regenerate-images-request.php]: The is_null check needs to happen almost right at the top as $attachment_id is used above this check which might cause errors/notices.

**Review by kloon — CHANGES_REQUESTED** [review]

File: includes/class-wc-regenerate-images-request.php (hunk 1 above)
Comment (bor0) [reply includes/class-wc-regenerate-images-request.php]: Ah good catch, totally missed that it's used before.

**Comment by mikejolley** [comment]
Leaving this open so we can look at a fallback.

**Comment by mikejolley** [comment]
https://github.com/woocommerce/woocommerce/pull/18640

**Pull request closed without merging**